sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+      ENV_IOT_HUB_HOSTNAME, NULL, false, iot_hub_hostname_span, &iot_hub_hostname_span));
+
+  // Paho requires that the MQTT endpoint be of the form ssl://<HUB ENDPOINT>:8883
+  AZ_RETURN_IF_FAILED(
+      create_mqtt_endpoint(mqtt_endpoint, (int32_t)sizeof(mqtt_endpoint), iot_hub_hostname_span));
+
+  // Initialize the hub client with the hub host endpoint and the default connection options
+  AZ_RETURN_IF_FAILED(az_iot_hub_client_init(
+      &client,
+      az_span_slice(iot_hub_hostname_span, 0, (int32_t)strlen(iot_hub_hostname)),
+      az_span_slice(device_id_span, 0, (int32_t)strlen(device_id)),
+      NULL));
+
+  return AZ_OK;
+}
+
+static az_result build_command_response_payload(
+    az_json_builder* json_builder,
+    az_span start_time_span,
+    az_span end_time_span,
+    az_span* response_payload)
+{
+  // Build the command response payload
+  AZ_RETURN_IF_FAILED(az_json_builder_append_begin_object(json_builder));
+  AZ_RETURN_IF_FAILED(
+      az_json_builder_append_property_name(json_builder, report_max_temp_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32(json_builder, (int32_t)device_max_temp));
+  AZ_RETURN_IF_FAILED(
+      az_json_builder_append_property_name(json_builder, report_min_temp_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32(json_builder, (int32_t)device_min_temp));
+  AZ_RETURN_IF_FAILED(
+      az_json_builder_append_property_name(json_builder, report_avg_temp_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32(json_builder, (int32_t)device_avg_temp));
+  AZ_RETURN_IF_FAILED(
+      az_json_builder_append_property_name(json_builder, report_start_time_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_string(json_builder, start_time_span));
+  AZ_RETURN_IF_FAILED(
+      az_json_builder_append_property_name(json_builder, report_end_time_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_string(json_builder, end_time_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_end_object(json_builder));
+
+  *response_payload = az_json_builder_get_json(json_builder);
+
+  return AZ_OK;
+}
+
+// Invoke the command requested from the service. Here, it generates a report for max, min, and avg
+// temperatures.
+static az_result invoke_getMaxMinReport(az_span payload, az_span response, az_span* out_response)
+{
+  // Parse the "since" field in the payload.
+  az_json_parser jp;
+  AZ_RETURN_IF_FAILED(az_json_parser_init(&jp, payload, NULL));
+  AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+  if (jp.token.kind != AZ_JSON_TOKEN_BEGIN_OBJECT)
+  {
+    return AZ_ERROR_PARSER_UNEXPECTED_CHAR;
+  }
+
+  // Get the user specified "since". Note that we don't honor this option to simplify the sample. It
+  // is merely passed on to the method call and return payload.
+  az_span start_time_span = AZ_SPAN_NULL;
+  while (jp.token.kind != AZ_JSON_TOKEN_END_OBJECT)
+  {
+    if (az_json_token_is_text_equal(&jp.token, report_command_payload_value_span))
```

Comment:
>if (az_json_token_is_text_equal(&jp.token, report_command_payload_value_span)) [](start = 4, length = 78)

This bit is confusing and we need to do a REALLY good job documenting this.  When the command request comes over, there is no "since" in it.  The "since" is just a hint to the DTML.  Instead all that we receive is the actual payload (e.g. an ISO8601 string in this case).  IoT Explorer knows about commands if you give it your models so you can see how it behaves with this by invoking this command through it.

Reply:
Okay changed. But I have some follow up questions